src/rkopenmdao/runge_kutta_scheme.py

```diff
@@ -4,18 +4,34 @@
 
 import numpy as np
 
-
 from rkopenmdao.butcher_tableau import ButcherTableau, EmbeddedButcherTableau
+from rkopenmdao.error_controller import ErrorController
+from rkopenmdao.error_controllers import Integral
 
 
 class RungeKuttaScheme:
-    """Implements functions used to apply a Runge-Kutta method a function represented by
+    """Implements functions used to apply a (embedded) Runge-Kutta method a function represented by
     a functor. The current main use is with an OpenMDAO problem wrapped to such a
-    functor."""
-
+    functor.
+
+    Parameters
+    ----------
+    butcher_tableau: ButcherTableau, EmbeddedButcherTableau
+        The butcher tableau to apply the Runge-Kutta method to.
+    stage_computation_functor: Callable[[np.ndarray, np.ndarray, float, float, float], np.ndarray]
+        Has the following form: (old_state, accumulated_stages, stage_time, delta_t, butcher_diagonal_element).
+    stage_computation_functor_jacvec: Callable[[np.ndarray, np.ndarray, float, float, float], np.ndarray]
+        Has the following form: (old_state_perturbation, accumulated_stages_perturbation,
+        stage_time, delta_t,butcher_diagonal_element).
+    stage_computation_functor_transposed_jacvec: Callable[[np.ndarray, float, float, float],Tuple[np.ndarray,np.ndarray]
+        Has the following form: (stage_perturbation, stage_time, delta_t, butcher_diagonal_element).
+    error_controller: ErrorController
+        Error controller that estimates the next time-difference jumps of a Runge-Kutta scheme.
+
+    """
     def __init__(
         self,
-        butcher_tableau: ButcherTableau,
+        butcher_tableau: ButcherTableau | EmbeddedButcherTableau,
```

Comment:
The type hint for `EmbeddedButcherTableau` is unnecessary. Since it derives from  `ButcherTableau`, an embedded one would pass a type check there anyway.